src/phonebook/Main.java

```diff
@@ -64,7 +67,9 @@ private static int pivot(List<Contact> contacts, int lo, int hi) {
     Contact pivot = contacts.get(lo);
     int j = lo + 1;
     for (int i = lo + 1; i <= hi; i++) {
-      if (pivot.getName().compareTo(contacts.get(i).getName()) >= 0) {
+      int comparing = Objects
```

Comment:
Now all the code looks better! You may also generalize your methods for sorting with <E> and then move the method in a separated class with sorting utils. But it is not required for this project. Just to have an additional practice.